hyperopt/spark.py

```diff
@@ -476,7 +477,9 @@ def run_task_on_executor(_):
                     result = domain.evaluate(params, ctrl=None, attach_attachments=False)
                     yield result
                 except BaseException as e:
-                    yield e
+                    error_string = traceback.format_exc()
+                    logger.error(error_string)
+                    yield type(e)(error_string)
```

Comment:
It is not guaranteed that you can reconstruct the error just using an error string. Does `e` already contain the stacktrace or can you use `with_traceback`?

Reply:
When pass `e` back to driver the traceback info is lost. We have to pass it by string.

Reply:
Did you try `with_traceback`?

Reply:
The issue is that is it not guaranteed that you can reconstruct the error this way. It might be okay to return the error without the traceback.

Reply:
We have a log as debug for the error message. https://github.com/hyperopt/hyperopt/blob/80b6850214f158ef966011a94bf95969d79e9b0c/hyperopt/spark.py#L367
We can change the level to show it to the users.

Reply:
@mengxr 
With 
```
tb = e.__traceback__
error_string = traceback.format_exception(type(e), e, tb)
```
Is that guaranteed to get the error message at least to the driver?

Reply:
the error message is always pickled.